### PR TITLE
[JBPM-9474] Produce events to Kafka by Signal or Messages Events

### PR DESCRIPTION
**Pull request by fjtirado**
Adding listener for signal and messages

**JIRA**: 

[JBPM-9474](https://issues.redhat.com/browse/JBPM-9474)

**depends on**: 
https://github.com/kiegroup/droolsjbpm-knowledge/pull/486

**merge with**
https://github.com/kiegroup/jbpm/pull/1808
https://github.com/kiegroup/droolsjbpm-integration/pull/2322

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java [hunk 1]
```diff
@@ -143,6 +145,29 @@ public void fireAfterSLAViolated(final ProcessInstance instance, NodeInstance no
...
     }
 
+    public void fireOnSignal(final ProcessInstance instance,
+                             NodeInstance nodeInstance,
+                             KieRuntime kruntime,
+                             String signalName,
+                             Object signalObject) {
+        if (hasListeners()) {
+            final SignalEvent event = new SignalEventImpl(instance, kruntime, nodeInstance, signalName, signalObject);
+            notifyAllListeners(event, (l, e) -> l.onSignal(e));
```
Comment (afalhambra) [inline drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java]: can be replaced with method reference
```suggestion
            notifyAllListeners(event, ProcessEventListener::onSignal);
```

File: drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java [hunk 2]
```diff
@@ -143,6 +145,29 @@ public void fireAfterSLAViolated(final ProcessInstance instance, NodeInstance no
...
+
+    public void fireOnMessage(final ProcessInstance instance,
+                              NodeInstance nodeInstance,
+                              KieRuntime kruntime,
+                              String messageName,
+                              Object messageObject) {
+        if (hasListeners()) {
+            final MessageEvent event = new MessageEventImpl(instance, kruntime, nodeInstance, messageName,
+                    messageObject);
+            notifyAllListeners(event, (l, e) -> l.onMessage(e));
```
Comment (afalhambra) [inline drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java]: same as above
```suggestion
            notifyAllListeners(event, ProcessEventListener::onMessage);
```

**Review by afalhambra — APPROVED** [review]
Just a couple of suggestions, but not really required.

File: drools-core/src/main/java/org/drools/core/event/MessageEventImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public String getMessageName() {
+        return messageName;
+    }
+
+    @Override
+    public Object getMessage() {
+        return messageObject;
+    }
```
Comment (gmunozfe) [inline drools-core/src/main/java/org/drools/core/event/MessageEventImpl.java:51]: _toString()_ method may be implemented to be similar to other classes that extend from ProcessEvent, wdyt?

File: drools-core/src/main/java/org/drools/core/event/SignalEventImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public String getSignalName() {
+        return signalName;
+    }
+
+    @Override
+    public Object getSignal() {
+        return signalObject;
+    }
```
Comment (gmunozfe) [inline drools-core/src/main/java/org/drools/core/event/SignalEventImpl.java:51]: Same here, implement _toString()_

**Review by gmunozfe — APPROVED** [review]
Looks good to me, a couple of comments to implement toString in event classes

File: drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java (hunk 1 above)
Comment (fjtirado) [reply drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java]: good point

File: drools-core/src/main/java/org/drools/core/event/MessageEventImpl.java (hunk 3 above)
Comment (fjtirado) [reply drools-core/src/main/java/org/drools/core/event/MessageEventImpl.java:51]: done

File: drools-core/src/main/java/org/drools/core/event/SignalEventImpl.java (hunk 4 above)
Comment (fjtirado) [reply drools-core/src/main/java/org/drools/core/event/SignalEventImpl.java:51]: done

**Review by afalhambra — APPROVED** [review]
LGTM

**Review by elguardian — APPROVED** [review]

**Pull request merged**